File: Programmers/스킬트리.js
function solution(skill, skill_trees) {
	var answer = 0;

	for (let user of skill_trees) {
		let cnt = 0;
		let idx = 0;
		for (let i of user) {
			if (skill.includes(i)) {
				if (skill.search(i) !== idx) {
					break;
				} else {
					idx += 1;
					cnt += 1;
				}
			} else {
				cnt += 1;
				continue;
			}
		}
		if (cnt === user.length) {
			answer += 1;
		}
	}

	return answer;
}

File: Programmers/스킬트리.py
def solution(skill, skill_trees):
    answer = 0
    
    for user in skill_trees:
        idx = -1
        cnt = 0
        for i in user:
            if i in skill:
                if idx+1 != skill.find(i):
                    break
                else:
                    idx = skill.find(i)
                    cnt += 1
            else:
                cnt += 1
                continue
        
        if cnt == len(user):
            answer += 1

    return answer